### PR TITLE
feat: show network name in binary versions

### PR DESCRIPTION
**Pull request by LexLuthr**
Fixes https://github.com/filecoin-project/boost/issues/1623 partially

This PR enhances the build command to build with network name

I have seen multiple times where users build a binary for calibnet or devnet and then try to use it for mainnet. This is mostly due to the fact that we don't show the network name in the `--version`.

```
% make debug             
rm -f boost
go build  -ldflags="-X=github.com/filecoin-project/boost/build.CurrentCommit=+git.8f88bde.dirty" -tags=debug -o boost ./cmd/boost
rm -f boostd
................
lexluthr@Lexs-Mac boost % ./boost --version
boost version 2.0.0-rc1+debug+git.8f88bde.dirty
lexluthr@Lexs-Mac boost % ./boostd-data --version
boostd-data version 1.4.0+mainnet+git.8f88bde.dirty
```

```
 make calibnet
Node.js version v16.13.2 is supported.
.............
rm -f boost
go build  -ldflags="-X=github.com/filecoin-project/boost/build.CurrentCommit=+git.8f88bde.dirty" -tags=calibnet -o boost ./cmd/boost
rm -f boostd
...................
lexluthr@Lexs-Mac boost % ./boost version 
No help topic for 'version'
lexluthr@Lexs-Mac boost % ./boost --version
boost version 2.0.0-rc1+calibnet+git.8f88bde.dirty
lexluthr@Lexs-Mac boost % ./boostd-data --version
boostd-data version 1.4.0+mainnet+git.8f88bde.dirty
```

```
make build             
Node.js version v16.13.2 is supported.
...............
rm -f boost
go build  -ldflags="-X=github.com/filecoin-project/boost/build.CurrentCommit=+git.8f88bde.dirty" -o boost ./cmd/boost
................
lexluthr@Lexs-Mac boost % ./boost --version
boost version 2.0.0-rc1+mainnet+git.8f88bde.dirty
lexluthr@Lexs-Mac boost % ./boostd-data --version
boostd-data version 1.4.0+mainnet+git.8f88bde.dirty
```

### CODE REVIEW

File: build/version.go [hunk 1]
```diff
@@ -1,9 +1,41 @@
...
+	BuildDebug        = 0x3
+	BuildCalibnet     = 0x4
+	BuildInteropnet   = 0x5
+	BuildButterflynet = 0x7
+)
+
+func BuildTypeString() string {
+	switch BuildType {
+	case BuildDefault:
+		return ""
```
Comment (nonsense) [inline build/version.go:19]:
```suggestion
```
Not sure what `default` stands for here, maybe just remove and bundle under `huh?`?

File: build/params_mainnet.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+//go:build !calibnet && !debug && 2k
+// +build !calibnet,!debug,2k
```
Comment (dirkmc) [inline build/params_mainnet.go:2]:
```suggestion
//go:build !calibnet && !debug && !2k
// +build !calibnet,!debug,!2k
```

**Review by dirkmc — COMMENTED** [review]
One correction but otherwise looks great 👍 

Let's also revert changes to go.mod and go.sum

**Review by dirkmc — COMMENTED** [review]
Could you please also undo the changes to the yugabyte directory as they will conflict with https://github.com/filecoin-project/boost/pull/1648

Otherwise LGTM 👍

**Review by dirkmc — APPROVED** [review]

**Pull request merged**